### ISSUE TITLE
Make it possible to choose a language

### ISSUE THREAD

**Issue by geecko86** [gcardone/junidecode#4]
In the case of Kanjis, your README says the following:

> The same character can be transliterated in different ways in different languages, for example the same ideogram has a different transliteration in Mandarin, Japanese and Korean. The choice in this case is: try to please most of the people (in the case of ideogram Chinese transliteration is always preferred since more than 1 billion people speak it and it is usually the root from which other languages derived).

There should be a way (and if there is, it's not documented) to specify in which language we want to transliterate a given string.

**Comment by gcardone** [comment]
This is by design:

1. This is a port of [Text::Unidecode](http://search.cpan.org/%7Esburke/Text-Unidecode-0.04/lib/Text/Unidecode.pm), which behaves in the same way.
1. Adding support for this means adding a knob to a currently settings-free library.
1. I am not a linguist, so even if I wanted to address this issue (and I am not convinced it's a good idea), I would need an actual linguist to help me.

This is currently infeasible.

**Comment by Morg-** [comment]
Hello GCardone,

I don't think you understand the problem here.
I'm not a linguist, but here's what happens when we use your library to translit actual Japanese text:
The Hiragana get transliterated to romaji correctly.
Then the Kanji get misinterpreted as chinese characters and get translit as chinese.

This gives you a romaji output which is half-japanese, half-chinese and 100% useless.
The same problem is sure to come up for Corean, or really any other language which uses chinese characters.

In other words, while I totally appreciate your intent to keep things as simple as possible, it's just that at this point, the library is simply pointless for anyone who doesn't simply want to translit Chinese.

Now of course, if your goal is not to provide a translit library that is actually usable in most languages, sorry for wasting your time.

Regards,


L.

**Comment by kberg** [comment]
No need to be nasty.

**Comment by geecko86** [comment]
Agreed, I'm not related to this person.

That said, I do agree with most of what he (@Morg-) was saying just now. 
I just think the wording could have used a bit more tact. :)

**Comment by gcardone** [comment]
JUnidecode is generic, and as such will make many people unhappy - but it tries to minimize global unhappiness at the expense of local unhappiness.

It looks like you know you are going to deal with Japanese text, so you can use more specialized libraries like [kuromoji](https://github.com/atilika/kuromoji), [kakasi](https://github.com/nicolas-raoul/kakasi-java), [jakaroma](https://github.com/nicolas-raoul/jakaroma), and [mecab](http://taku910.github.io/mecab/). You can easily read the input codepoints and decide whether to use one of these libraries of JUnidecode for transilteration.

I'm not going to make any of these a dependency of JUnidecode though - it is a simple, self-contained library.
